cpp/src/prims/per_v_random_select_and_transform_outgoine_e.cuh

```diff
@@ -0,0 +1,170 @@
+ * (if @p e_op does not access source property values). Use update_edge_src_property to fill the
+ * wrapper.
+ * @param edge_dst_value_input Wrapper used to access destination input property values (for the
+ * edge destinations assigned to this process in multi-GPU). Use either
+ * cugraph::edge_dst_property_t::view() (if @p e_op needs to access destination property values) or
+ * cugraph::edge_dst_dummy_property_t::view() (if @p e_op does not access destination property
+ * values). Use update_edge_dst_property to fill the wrapper.
+ * @param e_bias_op Quaternary (or quinary) operator takes edge source, edge destination, (optional
+ * edge weight), property values for the source, and property values for the destination and returns
+ * a graph weight type bias value to be used in biased random selection.
+ * @param e_op Quaternary (or quinary) operator takes edge source, edge destination, (optional edge
+ * weight), property values for the source, and property values for the destination and returns a
+ * value to be collected in the output. This function is called only for the selected edges.
+ * @param K Number of outgoing edges to select per (tagged-)vertex.
+ * @param with_replacement A flag to specify whether a single outgoing edge can be selected multiple
+ * times (if @p with_replacement = true) or can be selected only once (if @p with_replacement =
+ * false).
+ * @param invalid_value If @p invalid_value.has_value() is true, this value is used to fill the
+ * output vector for the zero out-degree vertices (if @p with_replacement = true) or the vertices
+ * with their out-degrees smaller than @p K (if @p with_replacement = false). If @p
+ * invalid_value.has_value() is false, fewer than @p K values can be returned for the vertices with
+ * fewer than @p K selected edges. See the return value section for additional details.
+ * @param do_expensive_check A flag to run expensive checks for input arguments (if set to `true`).
+ * @return std::tuple Tuple of an optional offset vector of type
+ * std::optional<rmm::device_uvector<size_t>> and a dataframe buffer storing the output values of
+ * type @p T from the selected edges. If @p invalid_value is std::nullopt, the offset vector is
+ * valid and has the size of @p frontier.size() + 1. If @p invalid_value.has_value() is true,
+ * std::nullopt is returned (the dataframe buffer will store @p frontier.size() * @p K elements).
+ */
+template <typename GraphViewType,
+          typename VertexFrontierBucketType,
+          typename EdgeSrcValueInputWrapper,
+          typename EdgeDstValueInputWrapper,
+          typename EdgeBiasOp,
+          typename EdgeOp,
+          typename T>
+std::tuple<std::optional<rmm::device_uvector<size_t>>,
+           decltype(allocate_dataframe_buffer<T>(size_t{0}, rmm::cuda_stream_view{}))>
+per_v_random_select_and_transform_outgoing_e(raft::handle_t const& handle,
```

Comment:
@ChuckHastings What do you think about `select_and_transfrom` vs `select_transform` in the primitive name?

I initially picked former but now getting more inclined to the latter as it is less verbose and more in line with other std:: thrust:: functions like `partitoin_copy`, `transform_reduce`, and so on. I guess removing `and` here won't cause much confusion?

I have a similar issue in naming `extract_and_transform_if...` primitives (a modification of the existing `extract_if_e` primitive to allow not just extracting an edge (source, destination, weight) triplets but to allow including edge IDs and types and so on... I feel like `extract_transform` may work better than `extract_and_transform`...

Reply:
I think removing the `_and` from the name is reasonable.  As you observe, the thrust model doesn't include and conjunctions, they are implied.